include/envoy/http/filter.h

```diff
@@ -76,6 +76,11 @@ class StreamFilterCallbacks {
   virtual uint64_t connectionId() PURE;
 
   /**
+   * @return const Network::Connection* the originating connection.
+   */
+  virtual const Network::Connection* connection() PURE;
```

Comment:
Can we please bring back all the logging, etc. changes you did in the other PR and mark connectionId() method deprecated and add to DEPRECATED.md? I would like to delete that method in 1.5.0 release cycle. Best if you could actually split this part of the change into a separate PR first so we can get that in and make this review less lines?